src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -108,7 +115,7 @@ CHIP_ERROR writeAdminsIntoFabricsListAttribute()
     CHIP_ERROR err = CHIP_NO_ERROR;
 
     // Loop through admins
-    int32_t fabricIndex = 0;
+    uint8_t fabricIndex = 0;
```

Comment:
OK, so the problem with doing this is that now the one-arg `writeFabricAttribute` call is broken on _all_ platforms, not just big-endian.  That call will treat the passed-in pointer as a pointer to `uint16_t`, but it's pointing to this one-byte value, so it will read some other byte from the stack in addition to this one byte and end up with who knows what for the list length.  But probably much larger than the actual length, and then bad things will happen.

We really need tests here, but we also really should not land this code in this shape (even with #8210 pending), imo.

The right fix is to have a `uint16_t` variable declared right above the one-arg `writeFabricAttribute` call, set its value to `fabricIndex`, and pass the pointer to _that_ variable, `reintrerpret_cast` to `uint8_t*`.